gradle/config/pmd/pmd.xml

```diff
@@ -0,0 +1,167 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<ruleset name="Base ruleset" xmlns="http://pmd.sourceforge.net/ruleset/2.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://pmd.sourceforge.net/ruleset/2.0.0 http://pmd.sourceforge.net/ruleset_2_0_0.xsd">
+
+    <description>
+        General Java quality rules.

```

Comment:
Some exclusion suggestions:

[Whitespace | WhitespaceAround] bio.terra.common.db.(DatabaseRetryUtils.java:16)
  '{' is not followed by whitespace.
  http://checkstyle.sourceforge.io/config_whitespace.html#WhitespaceAround
looks like there's some ways to limit it, to allow various types of things, but doesn't seem particularly useful to hit every reason to have whitespace around brackets.

style-ish things:
[Coding | DeclarationOrder] bio.terra.common.iam.(AuthenticatedUserRequest.java:91)
  Constructor definition in wrong order.
  http://checkstyle.sourceforge.io/config_coding.html#DeclarationOrder
[Metrics | JavaNCSS] bio.terra.common.kubernetes.(KubePodListener.java:82)
  NCSS for this method is 51 (max allowed is 50).
  http://checkstyle.sourceforge.io/config_metrics.html#JavaNCSS
​
[Coding | ReturnCount] bio.terra.common.kubernetes.(KubePodListener.java:82)
  Return count is 4 (max allowed for void methods/constructors/lambdas is 1).
  http://checkstyle.sourceforge.io/config_coding.html#ReturnCount
[Coding | ReturnCount] bio.terra.common.logging.(GoogleJsonLayout.java:220)
  Return count is 2 (max allowed for void methods/constructors/lambdas is 1).
  http://checkstyle.sourceforge.io/config_coding.html#ReturnCount
[Metrics | ClassFanOutComplexity] bio.terra.common.logging.(RequestLoggingFilter.java:45)
  Class Fan-Out Complexity is 22 (max allowed is 20).
  http://checkstyle.sourceforge.io/config_metrics.html#ClassFanOutComplexity

premature optimization suggestions
[Performance | InefficientStringBuffering] bio.terra.common.exception.(AbstractGlobalExceptionHandler.java:61)
  Avoid concatenating nonliterals in a StringBuffer/StringBuilder constructor or append().
  https://pmd.github.io/pmd-6.31.0/pmd_rules_java_performance.html#inefficientstringbuffering

[Code Style | LinguisticNaming] bio.terra.common.iam.(AuthenticatedUserRequest.java:46)
  Linguistics Antipattern - The setter 'setEmail' should not return any type except void linguistically
  https://pmd.github.io/pmd-6.31.0/pmd_rules_java_codestyle.html#linguisticnaming
[Design | NcssCount] bio.terra.common.kubernetes.(KubePodListener.java:84)
  The method 'run()' has a NCSS line count of 52.
  https://pmd.github.io/pmd-6.31.0/pmd_rules_java_design.html#ncsscount

[Design | CyclomaticComplexity] bio.terra.common.kubernetes.(KubePodListener.java:84)
  The method 'run()' has a cyclomatic complexity of 14.
  https://pmd.github.io/pmd-6.31.0/pmd_rules_java_design.html#cyclomaticcomplexity
​
[Error Prone | AvoidFieldNameMatchingMethodName] bio.terra.common.kubernetes.(KubeService.java:49)
  Field isShutdown has the same name as a method
  https://pmd.github.io/pmd-6.31.0/pmd_rules_java_errorprone.html#avoidfieldnamematchingmethodname

Reply:
Done